src/hamcrest/core/core/isinstanceof.py

```diff
@@ -11,21 +11,26 @@ class IsInstanceOf(BaseMatcher):
 
     def __init__(self, expected_type):
         if not is_matchable_type(expected_type):
-            raise TypeError('IsInstanceOf requires type')
+            raise TypeError('IsInstanceOf requires type or a tuple of classes and types')
         self.expected_type = expected_type
 
     def _matches(self, item):
         return isinstance(item, self.expected_type)
 
     def describe_to(self, description):
+        try:
+            type_description = self.expected_type.__name__
+        except AttributeError:
+            type_description = str(self.expected_type)
```

Comment:
I'd rather see a better description here, possibly `"one of %s" % ",".join(str(e) for e in self.expected_type)`